canvas_modules/common-canvas/src/common-properties/actions/button/button.jsx

```diff
@@ -27,10 +27,29 @@ class ButtonAction extends React.Component {
 	constructor(props) {
 		super(props);
 		this.state = {
+			icon: null
 		};
 		this.applyAction = this.applyAction.bind(this);
 	}
 
+	// Load icon via buttonIconHandler with a new type "actionButtonIcon" after mount
+	componentDidMount() {
+		const { controller, action } = this.props;
+		const buttonIconHandler = controller.getHandlers().buttonIconHandler;
+		if (buttonIconHandler && action) {
+			const propertyId = { name: action.data.parameter_ref };
```

Comment:
What is action.data isn't defined?  I would probably call this similar to 
```
actionHandler(this.props.action.name,
				this.props.controller.getAppData(), this.props.action.data)
```

Reply:
I don't think we'd pass `propertyId`.   I would think these are the values we'd pass in this case.  We'd have to make sure the documentation and typeScript is updated correctly for this.
```
const iconData = {
type: "actionButtonIcon",
buttonId: action.name,
data: action.data
};
```